src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs

```diff
@@ -304,8 +331,12 @@ internal static void WriteToStream(Stream input, Stream output, PSCmdlet cmdlet,
                     cmdlet.WriteProgress(record);
                 }
             }
-            catch (OperationCanceledException)
+            catch (OperationCanceledException exc)
```

Comment:
Are you sure this exception will be thrown when the cancellation happens?
In my script test, I get an `AggregateException` that wraps a `TaskCanceledException` instead.

```
$s = [System.Threading.CancellationTokenSource]::new()
try {
    $t = [System.Threading.Tasks.Task]::Delay(100000, $s.Token); ## delay for 100 seconds
    $s.CancelAfter(2000);   ## cancel after 2 seconds
    $t.Wait()
} catch {
    $e = $_
}
```
```
PS:32> $e.Exception.InnerException.GetType()

IsPublic IsSerial Name                                     BaseType
-------- -------- ----                                     --------
True     True     AggregateException                       System.Exception

PS:33> $e.Exception.InnerException.InnerExceptions[0].GetType()

IsPublic IsSerial Name                                     BaseType
-------- -------- ----                                     --------
True     True     TaskCanceledException                    System.OperationCanceledException
```

Reply:
No, there is no AggregateException. I guess because we use Wait(cts.Token) but in your example Wait().
![image](https://user-images.githubusercontent.com/22290914/189496414-e02d39e9-a8cd-4a10-bbd2-32a763980ba2.png)

Reply:
Hmm, we could fall in race condition if timeout was raised while we are in the loop body. So I replaced Wait with another overload which checks cancelation token first. https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs,2670

Reply:
I don't see any problem with timeout being raised while in the loop body. `Wait(millisecondsTimeout, cancellationToken)` will call [cancellationToken.ThrowIfCancellationRequested](https://source.dot.net/System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Threading/CancellationToken.cs.html#d2864d73804288fc) within [InternalWait](https://source.dot.net/System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs.html#2a1a81515e799ad9), after a quick `SpinWait`.